### PR TITLE
Simplify DVSNI

### PR DESCRIPTION
**Pull request by jdkasten**
This builds off of #179 and the previous discussion in #175.  It is also created as a separate branch within this repo to make it easier to edit.

### CODE REVIEW

File: draft-barnes-acme.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ normative:
   RFC2986:
   RFC3339:
   RFC3986:
+  RFC4366:
```
Comment (bifurcation) [inline draft-barnes-acme.md]: This should be RFC 6066

File: draft-barnes-acme.md [hunk 2]
```diff
@@ -1603,87 +1604,90 @@ failed.
...
-"\<nonce\>.acme.invalid".  The TLS server (i.e., the ACME client) should respond
-with a valid self-signed certificate containing both the domain name being
-validated and the domain name "\<Z\>.acme.invalid", where Z = SHA-256(R
-&#124;&#124; S).
-
-The ACME server's Challenge provides its random value R, and a random nonce used
-to identify the transaction:
+The Domain Validation with Server Name Indication (DVSNI) validation method
+proves control over a domain name by requiring the client to configure a TLS
+server to respond to specific connection attempts utilizing the Server Name
```
Comment (bifurcation) [inline draft-barnes-acme.md]: "a TLS server referenced by an A/AAAA record under the domain name"

File: draft-barnes-acme.md [hunk 3]
```diff
@@ -1603,87 +1604,90 @@ failed.
...
-   two dNSName values:
-  1. The domain name being validated
-  2. A name formed by hex-encoding Z and appending the suffix ".acme.invalid"
-5. Compute a nonce domain name by appending the suffix ".acme.invalid" to the
-   nonce provided by the server.
-6. Configure the TLS server such that when a client presents the nonce domain
-   name in the SNI field, the server presents the generated certificate.
+The client will generate a self-signed certificate with the subject's
+organizationName field set to the "signature" value from the JWS, i.e., the
+base64-encoded signature value and a subjectAlternativeName extension containing
```
Comment (bifurcation) [inline draft-barnes-acme.md]: s/value/value,/

File: draft-barnes-acme.md [hunk 4]
```diff
@@ -1603,87 +1604,90 @@ failed.
...
-The client's response provides its random value S:
+The response to the DVSNI challenge provides the validation JWS to the server.
 
 type (required, string):
 : The string "dvsni"
 
-s (required, string):
-: A random 32-byte secret octet string, Base64-encoded
+validation (required, string):
+: Base64-encoded SHA-256 hash of the DER-encoded challenge certificate
```
Comment (jdkasten) [inline draft-barnes-acme.md]: Forgot to update

Comment (bifurcation) [inline draft-barnes-acme.md]: This does not match what you've got below.  It doesn't seem like a bad idea to have the cert hash in the response, but you need to provide the JWS itself, either in the response or in the cert (probably the former).

File: draft-barnes-acme.md [hunk 5]
```diff
@@ -1603,87 +1604,90 @@ failed.
...
+4. Open a TLS connection to the domain name being validated on port 443,
+   presenting the value "\<token\>.acme.invalid" in the SNI field.
+5. Verify the following properties of the certificate provided by the TLS server:
+  * The organizationName field in the subject matches the "signature" value in
+    the "validation" JWS
+  * The certificate contains a single subjectAltName of the form
+    "\<token\>.acme.invalid".
 
 It is RECOMMENDED that the ACME server verify the challenge certificate using
 multi-path probing techniques to reduce the risk of DNS hijacking attacks.
```
Comment (bifurcation) [inline draft-barnes-acme.md]: This line should also be changed.  You don't "verify the certificate with multiple paths", you "perform validation TLS connections from multiple vantage points".

**Pull request merged**